packages/maskbook/src/plugins/NFT/utils.ts

```diff
@@ -0,0 +1,27 @@
+// nftUrl can be of these forms:
+//  - https://opensea.io/assets/<address>/<token_id>
+//  - https://app.rarible.com/token/<address>:<token_id>
+//
+// TODO: discuss about the superrare url format
+
+export function getRelevantUrl(textContent: string): URL {
+    return new URL('https://opensea.io/assets/0x1986f4c2e3ad5fe9778da67b1b836cf53b9e20cd/380/')
+}
+
+export function parseNftUrl(nftUrl: URL): [string, string] | any {
+    let address = null,
+        tokenId = null
+
+    if (nftUrl.hostname === 'opensea.io') {
+        let tokens = nftUrl.pathname.split('/')
+        address = tokens[tokens.length - 3]
+        tokenId = tokens[tokens.length - 2]
```

Comment:
```suggestion
    if (nftUrl.hostname === 'opensea.io' && /^\/assets\/(0x\w{42})\/(\w{70,})/.test(nftUrl.pathname)) {
        address = RegExp.$1
        tokenId = RegExp.$2
```

Reply:
Yeah, I was working on changing these to regex matchers.

Reply:
> parsed = new URL('https://opensea.io/assets/0x495f947276749ce646f68ac8c248420045cb7b5e/69540923947622736653594705474719122541146885144633674343119069962124023300097')
URL {
  href: 'https://opensea.io/assets/0x495f947276749ce646f68ac8c248420045cb7b5e/69540923947622736653594705474719122541146885144633674343119069962124023300097',
  origin: 'https://opensea.io',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'opensea.io',
  hostname: 'opensea.io',
  port: '',
  pathname: '/assets/0x495f947276749ce646f68ac8c248420045cb7b5e/69540923947622736653594705474719122541146885144633674343119069962124023300097',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
> /^\/assets\/(0x\w{40})\/(\w{70,})/.test(parsed.pathname)
true
> RegExp.$1
'0x495f947276749ce646f68ac8c248420045cb7b5e'
> RegExp.$2
'69540923947622736653594705474719122541146885144633674343119069962124023300097'
>

Reply:
plz verify user input correctness?

e.g: address MUST is SHA-1 strings (`0x[0-9a-fA-F]{40}`)
but regex `\w` include `_` / `-` / etc symbol, not can use it.

see https://mdn.io/RegExp#Regular_expression_and_Unicode_characters

Reply:
```js
export const urlRegexMatcher = /(https?:\/\/[^\s]+)/
export const openseaPathnameRegexMatcher = /^\/assets\/(0x[0-9a-fA-F]{40})\/(\w+)/
export const rariblePathnameRegexMatcher = /^\/token\/(0x[0-9a-fA-F]{40})\:(\w+)/
```

@septs does these look good to you?